spring-cloud-gcp-autoconfigure/src/test/java/com/google/cloud/spring/autoconfigure/pubsub/GcpPubSubAutoConfigurationTests.java

```diff
@@ -161,16 +175,77 @@ public void retryableCodes_many() {
 		ApplicationContextRunner contextRunner = new ApplicationContextRunner()
 				.withConfiguration(AutoConfigurations.of(GcpPubSubAutoConfiguration.class))
 				.withUserConfiguration(TestConfig.class)
-				.withPropertyValues("spring.cloud.gcp.pubsub.subscriber.retryableCodes=UNKNOWN,ABORTED,UNAVAILABLE,INTERNAL");
+				.withPropertyValues(
+						"spring.cloud.gcp.pubsub.subscriber.retryableCodes=UNKNOWN,ABORTED,UNAVAILABLE,INTERNAL");
 
 		contextRunner.run(ctx -> {
+			GcpPubSubProperties properties = ctx.getBean(GcpPubSubProperties.class);
+			GcpProjectIdProvider projectIdProvider = ctx.getBean(GcpProjectIdProvider.class);
+			DefaultSubscriberFactory defaultSubscriberFactory = ctx.getBean("defaultSubscriberFactory",
+					DefaultSubscriberFactory.class);
 
-			DefaultSubscriberFactory defaultSubscriberFactory = ctx.getBean("defaultSubscriberFactory", DefaultSubscriberFactory.class);
-			assertThat(FieldUtils.readField(defaultSubscriberFactory, "retryableCodes", true))
+			assertThat(properties.getSubscriber().getRetryableCodes())
+					.isEqualTo(new Code[] { Code.UNKNOWN, Code.ABORTED, Code.UNAVAILABLE, Code.INTERNAL });
+			assertThat(properties.computeRetryableCodes("subscription-name", projectIdProvider.getProjectId()))
+					.isEqualTo(new Code[] { Code.UNKNOWN, Code.ABORTED, Code.UNAVAILABLE, Code.INTERNAL });
+			assertThat(defaultSubscriberFactory.getRetryableCodes("subscription-name"))
+					.isEqualTo(new Code[] { Code.UNKNOWN, Code.ABORTED, Code.UNAVAILABLE, Code.INTERNAL });
+		});
+	}
+
+	@Test
+	public void retryableCodes_selectiveConfigurationSet() {
+		ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+				.withConfiguration(AutoConfigurations.of(GcpPubSubAutoConfiguration.class))
+				.withUserConfiguration(TestConfig.class)
+				.withPropertyValues(
+						"spring.cloud.gcp.pubsub.subscription.subscription-name.retryableCodes=UNKNOWN,ABORTED,UNAVAILABLE,INTERNAL");
+
+		contextRunner.run(ctx -> {
+
+			DefaultSubscriberFactory defaultSubscriberFactory = ctx.getBean("defaultSubscriberFactory",
+					DefaultSubscriberFactory.class);
+			assertThat(defaultSubscriberFactory.getRetryableCodes("subscription-name"))
```

Comment:
Add assert for a non-specified subscription here, too. Should be null, right?

Reply:
Good call. Yes, it will be null for a non-specified subscription. Added this check to the test below as well.